### PR TITLE
Organizing files to be more user-friendly

### PR DESCRIPTION
**Pull request by Heus-Sueh**
The changes I made:

- I introduced a new folder called "modules_config" to store all the "modules" configurations
- I introduced a new folder called "recipes" to store the "recipes"
- I introduced "main.yml" in "modules_config" to call all other "modules"
- I separated each module used in "recipe.yml" in its respective file in "modules_config"
- I added the "akmods", "fonts", "systemd" and "scripts" module to "modules_config"

Maybe I forgot to mention more changes

### CODE REVIEW

**Comment by xynydev** [comment]
I'm not so sure about this PR. The one thing I kind of like is the discoverability of new modules, but also see a _huge_ problem with that; new modules would have to be constantly added to the template and default configuration updated. That would create a constant new burden both for us as the maintainers of this project, and the maintainers of custom images who would need to merge those changes. Add that on top of the userbase handling the merge of this PR, when the last time recipe.yml in the template was changed some overwrote all of their changes accidentally. I don't know how many would be equipped to do a merge like this, most would probably be annoyed at it until discovering they _don't actually need to merge these changes and can keep on using their configuration, that already works well enough._

Then again, I think the default configuration should be minimal and easily understandable, with new features and modules being the users choice to include in their image. This repository structure also makes it harder to adopt a structure where modules are separated to different files based on what they add or where they are used, à la `common-packages.yml` or `firstboot.yml`. It is probably worth it, though, to include discussion on different repository structures on our new website https://blue-build.org/ (this project is going to be moved out of this GitHub org soon).

Thanks for [being bold](https://en.wikipedia.org/wiki/Wikipedia:Be_bold) anyway. I'll point this PR out to some other maintainers too, and you're free to defend your choices. My comment here didn't really even address the actual issues in your PR, only it's necessity.

File: config/recipes/recipe.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# image will be published to ghcr.io/<user>/<name>
+name: startingpoint
+# description will be included in the image's metadata
+description: A starting point for further customization of uBlue images. Make your own! https://ublue.it/making-your-own/
+
+# the base image to build on top of (FROM) and the version tag to use
+base-image: ghcr.io/ublue-os/silverblue-main
+image-version: 39 # latest is also supported if you want new updates ASAP
```
Comment (HexSleeves) [inline config/recipes/recipe.yml:8]: Does GH support templating files as well?

**Review by HexSleeves — APPROVED** [review]

File: config/modules_config/default-flatpaks.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+type: default-flatpaks # if included, https://github.com/ublue-os/yafti will be installed and set up
+notify: true # Send notification after install/uninstall is finished (true/false)
+#--------------------------------------------------------------------------------------------------#
+system:
+    # If no repo information is specified, Flathub will be used by default
+    repo-url: https://dl.flathub.org/repo/flathub.flatpakrepo
+    repo-name: flathub
+    repo-title: "Flathub (system-wide)" # Optional; this sets the remote's user-facing name in graphical frontends like GNOME Software
+    install:
+     # - Flatseal (Permission Manager): com.github.tchx84.Flatseal
```
Comment (fiftydinar) [inline config/modules_config/default-flatpaks.yml]: Format for this module is wrong.

It should be just

      # - com.github.tchx84.Flatseal
      
you can type that this is Flatseal (Permission Manager) adding additional #, like this:

      # - com.github.tchx84.Flatseal # Flatseal (Permission Manager).
      
Same goes for other entries too.

Comment (Heus-Sueh) [reply config/modules_config/default-flatpaks.yml]: fixed

**Comment by fiftydinar** [comment]
We redefined our defaults for the recipe.yml file recently & we concluded that we won't add much changes to it, to avoid potential merge conflict scenarios for the existing users.
Even if we introduced merge conflict which was very simple to solve, some users still had some problems in solving them.

So I don't agree with 5th point, adding additional modules like that, in current startingpoint.

For adopting this into new startingpoint (BlueBuild's template), I think that I agree that recipe should be minimal & stable, which is enough for most users.
Here's the link to BlueBuild's template repo:
https://github.com/blue-build/template

I like some changes, but they should be contributed in the repo mentioned above.
Prior discussion before making a PR is advisable.

Here's what I don't like:

For @xynydev's point about making it harder to include separated module files, I agree, as it adds another unnecessary (imo) layer of .yml files (you can simply include modules anyway in recipe.yml, no need for main.yml. I don't see the gain here).

And I forgot, I actually prefer this simple format for module_config yml's here:
```yaml
type: akmods
install:
  - nct6687d
  - openrazer
```

Than to use `description:`, `example:` etc. This info is unecessary imo, as this info is easily discoverable on website.

**Review by fiftydinar — APPROVED** [review]

**Review by fiftydinar — CHANGES_REQUESTED** [review]
I didn't want to approve whole PR, my mistake...

File: config/recipes/recipe.yml (hunk 1 above)
Comment (Heus-Sueh) [reply config/recipes/recipe.yml:8]: I don't know

**Comment by Heus-Sueh** [comment]
I agree with the point that @xynydev raised, one approach to this PR would be to put only the necessary modules, on the point of keeping everything in the recipe.yml this is good in one aspect, but it ends up becoming a big and messy file (not this is the case at the moment), in separate files it is better (at least for me) to browse and add new things, but perhaps the organization I made is too opinionated for my needs and not for a larger user base

**Comment by xynydev** [comment]
I like opinionation sometimes, but with this I am of the opinion that the template should being made to be as simply and stable as possible. I don't like the precedence this PR would set for the template being constantly updated when new modules are added or default configuration changed. I also think it would be better to keep the template pretty flat in terms of file structure, so that a user could find a structure that works best for their needs. It will always of course be possible to use this sort of multi-file structure when desired, but for simpler images like my own this feels very overkill.

---

I often like to make up examples of users wanting to do something to help my point in arguments like this, so here goes. Suppose this user reads a bare minimum amount of documentation for the project and just goes off of what they see being done in the template (hypothetically structured like in this PR) as the _correct_ way. They're first building a Gnome image with all of their small tweaks in the module-specific files. They then want to build a KDE image in addition, with a few tweaks to included packages, and flatpaks. They read somewhere that they can just make multiple recipes, so they make their `recipe-kde.yml` with the changes. They also need module changes, so they go down the chain following the example of the template and create `main-kde.yml`. They then need to make changes to only the `rpm-ostree` and `default-flatpaks` modules, so they go and create KDE versions of those files too, and change `main-kde.yml` to use them. Now for the differing parts of these image they have two separate sets of files they have to keep in sync. But aside from that, there's not really a problem with this setup, is there?

Well here's a clearer, more maintainable structure I would suggest instead. Once there's any intention of building multiple versions of the same image, I would first set up two recipes that both include a `common-modules.yml` file. After that point it is *up to the user* whether these modules are all in that file, or included from separate files in a subdirectory. Then for the differing parts, just add new instances of the same modules that already exist in `common-modules.yml` to either directly the recipe files or some files like `gnome.yml` and `kde.yml`.

That is the kind of structure that I feel is grown organically as people's repos grow in complexity. If the repos already start out with an include hierarchy, they will be likely more compelled to follow that structure, possibly leading to repos where the structure is planned for one purpose and used for another.

---

Just a quick summary of my thoughts.

Possible advantages of this structuring:
- One file for one module makes for an easier to follow config directory
- Individual module files have less complicated YAML nesting

Possible disadvantages of this structuring:
- Makes less clear how modules are just items in an array, meaning that they can be included multiple times and are executed in the order specified
- More complicated template might need more maintenance, thus more manual merges for people right where they hurt the most, in their configuration
- Users might be compelled to follow the structure rigidly even when they could prefer or be better off with some other structuring, something they made for themselves

---

*My opinions are 100% biased. Feel free to give me more reasons that this structure is better for the default template or argue with my objections. Now for the facts:*

This change will not be merged in this repository just due to the complicated merging. This could be the default under the new BlueBuild projects default template if so is together decided before the end of this month, when we'll have to move out from Ublue and announce BlueBuild officially.

**Comment by Heus-Sueh** [comment]
I'm going to close this PR because it only meets my needs

**Pull request closed without merging**